pkg/clusteragent/orchestrator/status.go

```diff
@@ -69,6 +69,39 @@ func GetStatus(ctx context.Context, apiCl kubernetes.Interface) map[string]inter
 		}
 	}
 	status["OrchestratorEndpoints"] = endpoints
+	setCacheInformationDCAMode(status)
+	setCollectionIsWorkingDCAMode(status)
+
+	// rewriting DCA Mode in case we are running in cluster check mode.
+	if orchestrator.KubernetesResourceCache.ItemCount() == 0 && config.Datadog.GetBool("cluster_checks.enabled") {
+		// we need to check first whether we have dispatched checks to CLC
+		stats, err := clusterchecks.GetStats()
+		if err != nil {
+			status["CLCError"] = err.Error()
+		} else {
+			// this and the cache section will only be shown on the DCA leader
+			if !stats.Active {
+				status["CLCEnabled"] = true
+				status["CollectionWorking"] = "Clusterchecks are activated but still warming up, the collection could be running on CLC Runners. To verify that we need the clusterchecks to be warmed up."
+			} else {
+				if _, ok := stats.CheckNames[orchestrator.CheckName]; ok {
+					status["CLCEnabled"] = true
+					status["CacheNumber"] = "No Elements in the cache, since collection is run on CLC Runners"
```

Comment:
What this cache is used for ?

Reply:
Every `n` (default 10) seconds we retrieve values from the informer. 
We use an internal cache to decide whether the retrieved information is new or not, if they are "new" we do our parsing (secret scrubbing, conversion ...), else we skip.